modules/reindex/src/main/java/org/elasticsearch/index/reindex/BulkIndexByScrollResponseContentListener.java

```diff
@@ -41,6 +43,12 @@ protected RestStatus getStatus(R response) {
                 status = failure.getStatus();
             }
         }
+        for (ShardSearchFailure failure: response.getSearchFailures()) {
+            RestStatus failureStatus = ExceptionsHelper.status(failure.getCause());
+            if (failureStatus.getStatus() > status.getStatus()) {
```

Comment:
This is kind of a weird way to order failures, but I guess it works. I think adding a comment might help others in the future in understanding your thinking